src/main/java/io/netty/incubator/codec/quic/QuicConnectionCloseEvent.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.quic;
+
+import java.util.Arrays;
+
+/**
+ * Event that is generated if the remote peer sends a CLOSE_CONNECTION. This allows to inspect the
+ * {@link #error()} and so see if the close was done because of an error or not.
+ */
+public final class QuicConnectionCloseEvent implements QuicEvent {
+
+    final boolean applicationClose;
+    final int error;
+    final byte[] reason;
+
+    QuicConnectionCloseEvent(boolean applicationClose, int error, byte[] reason) {
+        this.applicationClose = applicationClose;
+        this.error = error;
+        this.reason = reason;
+    }
+
+    /**
+     * Return {@code true} if this was an application close, {@code false} otherwise.
+     *
+     * @return  if this is an application close.
+     */
+    public boolean isApplicationClose() {
+        return applicationClose;
+    }
+
+    /**
+     * Return the error that was provided for the close.
+     *
+     * @return the error.
+     */
+    public int error() {
+        return error;
+    }
+
+    /**
+     * Returns {@code true} if a <a href="https://www.rfc-editor.org/rfc/rfc9001#section-4.8">TLS error</a>
+     * is contained.
+     * @return {@code true} if this is an {@code TLS error}, {@code false} otherwise.
+     */
+    public boolean isTlsError() {
+        return !applicationClose && error >= 0x0100;
+    }
+
+    /**
+     * Return the reason for the close.
+     *
+     * @return  the reason.
+     */
+    public byte[] reason() {
+        return reason.clone();
```

Comment:
Pretty vague what this is. Quick look through the Quiche code didn't tell me much either.